### PR TITLE
Add versioned deprecation warning

### PR DESCRIPTION
**Pull request by irenedea**
This allows us to track deprecations and know immediately when they should be removed.

### CODE REVIEW

File: composer/utils/warnings.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    Attributes:
+        message (str): The deprecation message describing why the feature is deprecated.
+        remove_version (str): The version in which the feature will be removed.
+
+    Example:
+        >>> def deprecated_function():
+        ...     warnings.warn(
+        ...         VersionedDeprecationWarning(
+        ...             "Function XYZ is deprecated.",
+        ...             after_version="2.0.0"
```
Comment (mvpatel2000) [inline composer/utils/warnings.py:21]:
```suggestion
        ...             remove_version="2.0.0",
```

**Review by mvpatel2000 — COMMENTED** [review]
Smol unit test to try warning?

**Comment by irenedea** [comment]
@mvpatel2000 added

**Review by mvpatel2000 — APPROVED** [review]
TY!

**Comment by mvpatel2000** [comment]
@irenedea  can we merge this?

**Review by mvpatel2000 — APPROVED** [review]
Thanks!

**Pull request merged**